support-frontend/assets/components/button/_sharedButton.jsx

```diff
@@ -59,6 +66,7 @@ const SharedButton = ({
   return createElement(element, {
     className,
     ref: getRef,
+    id: postDeploymentTestID ? `qa-${postDeploymentTestID}` : null,
```

Comment:
nice neat idea doing it at this point (when the type is lost, add the type information into the value)